### DISCUSSION TITLE
Webbhook initiates full scan even for files outside of filter scope

### DISCUSSION THREAD

**Discussion by Weissrolf** [abraunegg/onedrive#2136] [Q&A] [answered]
Hello,

I noticed that the webhook initiates a full scan of all sync folder even when a file outside of the client's scope is update via webhook. What I mean is that the onedrive client is set to use --single-directory, but when the webhook reports a file being updates *outside* the scope of --single-directory then the client still rescans the --single-directory structure for changes.

Example:
PC 1 --monitor --single-directory 'onedrive/folder 1'
PC 2 changes a file in 'onedrive/folder 2'
PC 1 receives webhook message and scans 'onedrive/folder 1' recursively for changes, despite no changes being made to this scope.

I mostly wonder if a complete scan is mandatory with webhooks, are they so unspecific? I expected the webhook to report exactly which files were changed remotely and then the client to only sync these specific files instead of rescanning the whole structure.

PS: I switched from skip_dir and sync_list to single-directory, because the former seems to be much slower in skipping excluded folders compared to the latter.

**Comment by abraunegg** [comment, marked as answer]
@Weissrolf 

> I mostly wonder if a complete scan is mandatory with webhooks, are they so unspecific?

The issue here is understanding the Microsoft OneDrive architecture.

Their architecture assumes that what is 'online' is what will be 'local' .. verbatum .. exact replica.

This client, has 'client side filtering' so that you only store locally what you want to store - for example your `PC 1 --monitor --single-directory 'onedrive/folder 1'` item

OneDrive has zero concept of this, thus, a webhook is a trigger to go get the latest changes from OneDrive .. this is why you are seeing this behavior and there is zero way to change this - as this would require OneDrive to have the concept | notion of your client side configurations ..... and that is never going to happen.

**Comment by Weissrolf** [comment]
So the webhook does not report specific file changes to the client, but only "I changed, better rescan everything" (regardless of local filtering)?

**Reply by abraunegg** [reply]
@Weissrolf 
Correct ... OneDrive simply notifies a change occured ...

**Reply by Weissrolf** [reply]
Thanks for the explanation and for the useful software (running on a cloud VM here to sync with my local NAS which runs the backups).
